### DISCUSSION TITLE
Migrating forms between environments?

### DISCUSSION THREAD

**Discussion by lbriet** [solspace/craft-freeform#221] [Questions & Answers] [answered]
Related to https://github.com/solspace/craft3-freeform/discussions/121 - we have a use-case where we are working on a project with multiple developers. Forms play a large role therein, and without the ability to sync forms/fields to .yaml config files, parallel development is turning into a database juggling-act.

That _seems_ to be feasible by ex-/im-porting the 25 `freeform_..` DB tables exclusively.

Before we continue much further down that path though - is that actually correct? Is everything (free)form related fully contained within those 25 tables, or are we going to risk diverging/mismatching DBs elsewhere?

**Comment by gustavs-gutmanis** [comment, marked as answer]
Hi @lbriet,

All of the tables except `freeform_submissions` and `freeform_export_settings` are self-contained and should not cause you any trouble exporting/importing.
`freeform_submissions` is a table which links to the `craft_elements` table and will require the ID's to match.
`freeform_export_settings` is a table which links to the `craft_users` table and will require `freeform_export_settings.userId => craft_users.id` to match.

**Reply by lbriet** [reply]
Thanks @gustavs-gutmanis

Seems fine to continue as we are then (ideally, without any form submissions while we're in development).

Appreciate the help!
